V1/Services/SessionService.cs

```diff
@@ -21,18 +20,18 @@ public SessionService(IWalletService walletService, ISessionRepository sessionRe
             _sessionSettings = sessionSettings;
         }
 
-        public async Task<Session> CreateSession(CreateSessionRequest createSessionRequest)
+        public async Task<Session> CreateSession(string mobileNumber)
         {
             var walletRequest = new CreateWalletRequest
             {
-                MobileNumber = createSessionRequest.MobileNumber,
-                MobileNumberReference = createSessionRequest.MobileNumberReference
+                MobileNumber = mobileNumber
             };
-            var wallet = await _walletService.AddWalletToDatabase(walletRequest);
+            var wallet = await _walletService.CreateWallet(walletRequest);
```

Comment:
Wallet gets created twice. Once in the Organisation service in the MobileEntry method and then again here. So the session is only linked to the 2nd wallet. Rather just pass the wallet into this method.